### DISCUSSION TITLE
Combining Two Trained Models

### DISCUSSION THREAD

**Discussion by Rushit-Jain** [explosion/spaCy#8521] [Help: Best practices]
I am a bit of a newbie to spaCy, so have been confused over a few things. I was wondering if there's a way to combine two trained NER models. Both the models are recognising the same entities, just that they are trained on two different parts of a large corpus.

1. The first model is trained on recognizing the entities of a large corpus, while the second one is also the same. The only difference is, they're trained on two parts of the corpus and hence, might not be very accurate.

2. My question - Is there a way to combine or ensemble these two models together? And can it help increasing the overall accuracy?

**Comment by polm** [comment]
We actually just released [a tutorial about combining two NER models](https://github.com/explosion/projects/tree/v3/tutorials), though it's for the case where they cover *different* entities. 

> My question - Is there a way to combine or ensemble these two models together? And can it help increasing the overall accuracy?

In spaCy you cannot ensemble these methods in a straightforward way. You can put one before the other so one might pick up entities the other missed, but if they're predicting the same thing that seems unlikely to work. 

In theory you could concat the vectors of the models for predictions or something, but spaCy won't help you do that. It's also hard to say more about it without knowing more about your models. When you say they're trained on two parts of the corpus, are those random slices, or are they chosen in some principled manner, even if the list of entities is the same?

I suspect you're better off just training a single model, but it really is hard to say much without knowing anything about your data.

**Reply by Rushit-Jain** [reply]
The corpus was sliced randomly into two parts of around 1000 docs each. Consider they are almost equal in size. The list of entities is same though

**Reply by polm** [reply]
Yeah you should just train one model on all the data then.

Why did you train the models that way?

**Reply by Rushit-Jain** [reply]
A lack of resources I would say. Haven't been able to find a good workspace. I am still trying solutions though. Just wanted to know if it would work this way. Thanks a ton

**Reply by liverpoolka** [reply]
hi! have a similar question - how we can combine two training json files?

**Reply by polm** [reply]
Hey @almazhankapan,

It's possible to combine two training json by manipulating the json to append the relevant data, but it's kind of complicated so I wouldn't recommend it.

What you can do is convert each json to a `.spacy` file individually and then put those `.spacy` files in a directory together. When you specify a path to training data, if you give a directory all `.spacy` files in it will be used.

**Reply by liverpoolka** [reply]
thanks a lot!
